src/interrupt.c

```diff
@@ -76,26 +76,28 @@ extern __inline__ int metal_interrupt_enable(struct metal_interrupt *controller,
 extern __inline__ int
 metal_interrupt_disable(struct metal_interrupt *controller, int id);
 
-extern __inline__ unsigned int
-metal_interrupt_get_threshold(struct metal_interrupt *controller);
-
 extern __inline__ int
-metal_interrupt_set_threshold(struct metal_interrupt *controller,
-                              unsigned int threshold);
+metal_interrupt_vector_enable(struct metal_interrupt *controller, int id);
 
-extern __inline__ unsigned int
-metal_interrupt_get_priority(struct metal_interrupt *controller, int id);
+extern __inline__ int
+metal_interrupt_vector_disable(struct metal_interrupt *controller, int id);
 
 extern __inline__ int
-metal_interrupt_set_priority(struct metal_interrupt *controller, int id,
```

Comment:
It looks like the `get_priority` and `set_priority` methods are getting deleted and not replaced?

Reply:
Yes, the first patch in this PR removes four APIs, because these four APIs are defined twice in the interrupt.c.